packages/iTwinUI-react/src/core/Modal/Modal.tsx

```diff
@@ -95,137 +49,30 @@ export type ModalProps = {
-      overlayRef.current?.focus();
-    } else {
-      previousFocusedElement.current?.focus();
-    }
-    const modalOverlayRef = overlayRef.current;
-    return () => {
-      modalOverlayRef?.contains(document.activeElement) &&
-        previousFocusedElement.current?.focus();
-    };
-  }, [isOpen]);
-
-  React.useEffect(() => {
-    if (!ownerDocument) {
-      return;
-    }
-
-    if (isOpen) {
-      originalBodyOverflow.current = ownerDocument.body.style.overflow;
-      ownerDocument.body.style.overflow = 'hidden';
-    } else {
-      ownerDocument.body.style.overflow = originalBodyOverflow.current;
-    }
-    return () => {
-      ownerDocument.body.style.overflow = originalBodyOverflow.current;
-    };
-  }, [isOpen, ownerDocument]);
-
-  const handleKeyDown = (event: React.KeyboardEvent) => {
-    // Prevents React from resetting its properties
-    event.persist();
-    if (isDismissible && closeOnEsc && event.key === 'Escape' && onClose) {
-      onClose(event);
-    }
-    if (onKeyDown) {
-      onKeyDown(event);
-    }
-  };
-
-  const handleMouseDown = (event: React.MouseEvent) => {
-    // Prevents React from resetting its properties
-    event.persist();
-    if (event.target !== overlayRef.current) {
-      return;
-    }
-    if (isDismissible && closeOnExternalClick && onClose) {
-      onClose(event);
-    }
-  };
-
-  return !!container ? (
-    ReactDOM.createPortal(
-      <CSSTransition
-        in={isOpen}
-        classNames='iui-modal-animation'
-        timeout={{ exit: 600 }}
-        unmountOnExit={true}
-      >
-        <FocusTrap>
-          <div
-            className={cx(
-              'iui-modal',
-              { 'iui-modal-full-page': styleType === 'fullPage' },
-              { 'iui-modal-visible': isOpen },
-              className,
-            )}
-            tabIndex={-1}
-            onKeyDown={handleKeyDown}
-            ref={overlayRef}
-            onMouseDown={handleMouseDown}
-            {...rest}
-          >
-            <div
-              className='iui-modal-dialog'
-              id={id}
-              style={style}
-              role='dialog'
-              aria-modal='true'
-            >
-              <div className='iui-title-bar'>
-                <div className='iui-title'>{title}</div>
-                {isDismissible && (
-                  <IconButton
-                    size='small'
-                    styleType='borderless'
-                    onClick={onClose}
-                    aria-label='Close'
-                  >
-                    <SvgClose />
-                  </IconButton>
-                )}
-              </div>
-              {children}
-            </div>
-          </div>
-        </FocusTrap>
-      </CSSTransition>,
-      container,
-    )
-  ) : (
-    <></>
+  return (
+    <Dialog.Backdrop
+      isDismissible={isDismissible}
+      dialogRootId={modalRootId}
+      {...rest}
+    >
+      <Dialog style={style} id={id} aria-modal>
```

Comment:
api looks kinda weird. `Dialog.Backdrop` is a parent of the main `Dialog` component?

Reply:
It's the same in css `iui-dialog-backdrop`-> `iui-dialog`

Reply:
The user won't care about our CSS classnames. My concern here is just about the potentially confusing.

I was looking for other examples and found these:
- [`radix-ui`](https://www.radix-ui.com/docs/primitives/components/dialog#anatomy) uses `Dialog.Overlay` and `Dialog.Content` with an additional `Dialog.Root` as the main wrapper.
- [`reach-ui`](https://reach.tech/dialog#dialogoverlay) uses `DialogOverlay` and `DialogContent`.
- [`ariakit`](https://reakit.io/docs/dialog/#backdrop) uses `Dialog` and `DialogBackdrop` (similar to ours but without `.`)
- [`mui`](https://mui.com/material-ui/api/dialog/#props) has `Dialog` and `Backdrop` as separate components and `Dialog` has a `BackdropComponent` prop.

Reply:
Yeah, I also don't really like how it is right now but I don't know how to name `Dialog`. Maybe `Dialog.Dialog` but also looks weird, `Dialog.Content` is also taken.

Reply:
`Dialog.Main`?

Reply:
But if we intend for the users to use just plain `Dialog` then using some subcomponent kind of feels odd to me.

Reply:
Yeah, you can use just the `Dialog` but then you need to handle everything by yourself.

Reply:
What if we make `Dialog` root component that has almost the same props as `Modal`, but it would have a prop for toggling backdrop or additional type. Because I checked other libs and most of them has a boolean prop for backdrop or if they have a subcomponent, it is placed as a sibling for dialog content.

Reply:
Dropped `Dialog.Main`, now the `Dialog` takes care for that and having `backdrop` prop where users can pass custom backdrop if they wish.